src/Rector/Deprecation/DrupalSetMessageRector.php

```diff
@@ -77,7 +79,7 @@ public function refactor(Node $node): ?Node
     {
         /** @var Node\Expr\FuncCall $node */
         if ($this->getName($node->name) === 'drupal_set_message') {
-            $class = $this->betterNodeFinder->findParentType($node, Node\Stmt\Class_::class);
+            $class = $this->findParentType($node, Node\Stmt\Class_::class);
             if ($this->checkClassTypeHasTrait($class, 'Drupal\Core\Messenger\MessengerTrait')) {
```

Comment:
Know what'd be great? I wonder if Rector got "smart" enough to not add a trait which already exists in the class. So we don't even need to do this. I don't know where to even check that in the Rector source... But it's an idea

Reply:
If you're in a class knowing if a trait exists is trivial though... or am I misunderstanding?

Reply:
It is trivial. The code is fine. I was just curious if maybe Rector has code built in to avoid duplicate trait definitions.

Reply:
Actually. I think I saw fqcn as key in a list of traits. So it might